recipes/dpctl/meta.yaml

```diff
@@ -0,0 +1,92 @@
+{% set required_compiler_version = "2024.2.0" %}
+
+{% set pyproject = load_file_data('pyproject.toml') %}
+{% set py_build_deps = pyproject.get('build-system', {}).get('requires', []) %}
+
+{% set version = "0.20.2" %}
+{% set buildnumber = "0" %}
+
+package:
+    name: dpctl
+    version: {{ version }}
+
+source:
+    url: https://github.com/IntelPython/dpctl/archive/refs/tags/{{ version }}.tar.gz
+    sha256: ad14bb214e341d9315245012e39d2aec6ca8217f99114790134664420d0eae4f
+
+build:
+    number: {{ buildnumber }}
+    ignore_run_exports:
+        - level-zero
+    missing_dso_whitelist:
+        - '**/DPCTLSyclInterface.dll'  # [win]
+    skip: True  # [osx or py<39 or py>313]
+
+requirements:
+    # TODO: keep in sync with /pyproject.toml
+    build:
+        - {{ compiler('cxx') }}
+        - {{ stdlib('c') }}
+        - {{ compiler('dpcpp') }} >={{ required_compiler_version }}
+    host:
+        - python
+        - pip >=24.0
+        - level-zero-devel >=1.16
+        - pybind11 >=2.12
+        - {{ pin_compatible('intel-sycl-rt', min_pin='x.x', max_pin='x') }}
+        - {{ pin_compatible('intel-cmplr-lib-rt', min_pin='x.x', max_pin='x') }}
+        # Ensure we are using latest version of setuptools, since we don't need
+        # editable environments for release.
+        - setuptools >=69
+        {% for dep in py_build_deps %}
+        {% if dep.startswith('ninja') %}
+        - {{ dep.split(';')[0] }}  # [not win]
+        {% elif dep.startswith('cmake') %}
+        - {{ dep }}
+        {% elif dep.startswith('build>=') %}
+        - {{ 'python-' ~ dep }}
+        {% elif dep.startswith('cython') %}
+        {% if dep.split(';')[1] == "python_version<'3.13'" %}
+        - {{ dep.split(';')[0] }}  # [py<313]
+        {% else %}
+        - {{ dep.split(';')[0] }}  # [py>=313]
+        {% endif %}
```

Comment:
Can you remove cmake, build, etc from `py_build_deps` jinja var and add these manually. The current code is really hard to read.